platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/StateGarbageCollector.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.signed;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Instant;
+
+/**
+ * This component is responsible for the deletion of signed states. In case signed state deletion is expensive, we never
+ * want to delete a signed state on the last thread that releases it.
+ */
+public interface StateGarbageCollector {
+
+    /**
+     * Register a signed state with the garbage collector. The garbage collector will eventually delete the state when
+     * it is no longer needed.
+     *
+     * @param state the state to register
+     */
+    void registerState(@NonNull ReservedSignedState state);
+
+    /**
+     * This method is called periodically to give the signed state manager a chance to delete states.
+     *
+     * @param now the current time
+     */
+    void heartbeat(@NonNull Instant now);
```

Comment:
FYI - it isn't strictly *required* to have `now` as a parameter, with the recent changes to proxy wiring to permit parameterless bound methods.

Reply:
Right, forgot about that feature. Simplified.